### PR TITLE
Syndieship turrets target non-nukie cyborgs

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Syndieship turrets now target non emagged or non nukie cyborgs. IDK if this is a fix or a rebalance coz from the way the original code reads it SEEMS like the original coder intended for the turrets to target them but didnt go all the way through (maybe)

## Why It's Good For The Game

Cyborgs now must go through the same hardships normal people do when trying to invade the nukeops ship

## Changelog

:cl:
balance: Syndieship turrets now target non emagged or non nukie cyborgs
/:cl:

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -448,8 +448,8 @@ DEFINE_BITFIELD(turret_flags, list(
 
 			if(iscyborg(sillycone))
 				var/mob/living/silicon/robot/sillyconerobot = A
-				if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
-					continue
+				if((ROLE_SYNDICATE in faction) && !sillyconerobot.emagged)
```
[the remark this answers is not in the document]
Comment (SmoSmoSmoSmok) [reply code/game/machinery/porta_turret/portable_turret.dm:451]: THIS works and is tested

**Comment by SmoSmoSmoSmok** [comment]
> I like how it forces nukies to decide between protection from being raided by station borgs, or being able to purchase syndicate borgs, it's small things like this that make alternatives (like Cayenne) worth getting.

they still get the danger of being raided, the raiders just must go through some extra steps. cayenne is a bitch itll die to 1 welder from an engieborg

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fikou) [reply code/game/machinery/porta_turret/portable_turret.dm:451]: dont remove the lazylen

[the remark this answers is not in the document]
Comment (SmoSmoSmoSmok) [reply code/game/machinery/porta_turret/portable_turret.dm:451]: my point was borgs are targetted by default roundstart this way but alright

**Comment by Sealed101** [comment]
cayenne is a force to be reckoned with y'all just sleeping
willard's point is valid, but the amount of bullshit station borgs have pulled on the nukies over the years without any effort is, while extremely funny, kind of absurd. it's also a pain in the ass to set every turret manually to target borgs, since syndie infiltrator doesn't have a turret control panel. i do it but that doesn't mean anyone else does. either put a control panel on the ship without any distinction between silicon friend or foe, or set them on by default to target non-syndie borgs, i guess

**Comment by SkeletalElite** [comment]
What are you people on the only thing good about cayenne is giving them a macro bomb and sending them on a suicide mission

**Comment by IndieanaJones** [comment]
I'm almost certain we did this before and had to revert it because something bad happened.  I don't remember what, but I think it might have to do something with it shooting nukie borgs under some circumstances.

**Comment by ATH1909** [comment]
> I'm almost certain we did this before and had to revert it because something bad happened. I don't remember what, but I think it might have to do something with it shooting nukie borgs under some circumstances.

emagged borgs, maybe?

it'd be nice if we gave the syndicate ship a turret control panel thingie so that the ops could choose whether or not they want their turrets to target station-aligned borgs (if they're planning to, say, emag them).

**Comment by Serxule** [comment]
> it'd be nice if we gave the syndicate ship a turret control panel thingie so that the ops could choose whether or not they want their turrets to target station-aligned borgs (if they're planning to, say, emag them).

that was actually how it used to be, i have no idea when or why it was removed however

**Comment by Sealed101** [comment]
you sir are incorrect

> What are you people on the only thing good about cayenne is giving them a macro bomb and sending them on a suicide mission

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -448,8 +448,8 @@ DEFINE_BITFIELD(turret_flags, list(
 
 			if(iscyborg(sillycone))
 				var/mob/living/silicon/robot/sillyconerobot = A
-				if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
-					continue
+				if((ROLE_SYNDICATE in faction) && !sillyconerobot.emagged && LAZYLEN(faction))
+					targets+= sillyconerobot
```
Comment (Rohesie) [inline code/game/machinery/porta_turret/portable_turret.dm:452]: The feature is good, but take a look at the `if(issilicon(A))` check.
This code is confusing, somewhat duplicated and I'm not entirely sure it does what you intend it to.
Have you tested this?

**Review by Rohesie — CHANGES_REQUESTED** [review]
Do please test your changes.

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 2 above)
Comment (SmoSmoSmoSmok) [reply code/game/machinery/porta_turret/portable_turret.dm:452]: > The feature is good, but take a look at the `if(issilicon(A))` check. This code is confusing, somewhat duplicated and I'm not entirely sure it does what you intend it to. Have you tested this?

yep its all tested and WORKS as intended. Only cyborgs that arent emagged or arent part of the synditeam will be shot down

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 3]
```diff
@@ -439,17 +439,15 @@ DEFINE_BITFIELD(turret_flags, list(
...
-			if(sillycone.stat || in_faction(sillycone))
-				continue
-
 			if(iscyborg(sillycone))
 				var/mob/living/silicon/robot/sillyconerobot = A
-				if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
+				if((turret_flags & TURRET_FLAG_SHOOT_BORGS) && sillyconerobot.stat != DEAD)
+					targets += sillyconerobot
+					continue
+				if(sillyconerobot.stat || in_faction(sillyconerobot))
```
Comment (Rohesie) [inline code/game/machinery/porta_turret/portable_turret.dm:447]:
```suggestion
				if(sillyconerobot.stat != CONSCIOUS)
					continue
				if((turret_flags & TURRET_FLAG_SHOOT_BORGS))
					targets += sillyconerobot
					continue
				if(in_faction(sillyconerobot))
```
Early continue.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -439,17 +439,15 @@ DEFINE_BITFIELD(turret_flags, list(
...
 					continue
+				if((ROLE_SYNDICATE in faction) && !sillyconerobot.emagged && LAZYLEN(faction))
```
Comment (Rohesie) [inline code/game/machinery/porta_turret/portable_turret.dm:449]:
```suggestion
				if((ROLE_SYNDICATE in faction) && !sillyconerobot.emagged)
```
`X in null` should still be false, no? No need to check whether the list is populated.

Comment (SmoSmoSmoSmok) [reply code/game/machinery/porta_turret/portable_turret.dm:449]: That's what I did at first but JohnWillard told to include the LAZYLEN(faction) when he reviewed this. I hate him

**Pull request merged**